packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts

```diff
@@ -196,14 +196,29 @@ export const createRdsLambda = (
  * add the name. We can figure out the right way to expose this to customers if needed, but for now we are not invoking `setResourceName`
  * because it would have no effect.
  */
-export const createLayerVersionCustomResource = (scope: Construct, resourceNames: SQLLambdaResourceNames): AwsCustomResource => {
+export const createLayerVersionCustomResource = (
+  scope: Construct,
+  resourceNames: SQLLambdaResourceNames,
+  context: TransformerContextProvider,
+): AwsCustomResource => {
   const { SQLLayerManifestBucket, SQLLayerManifestBucketRegion, SQLLayerVersionManifestKeyPrefix } = ResourceConstants.RESOURCES;
 
   const key = Fn.join('', [SQLLayerVersionManifestKeyPrefix, Fn.ref('AWS::Region')]);
 
   const manifestArn = `arn:aws:s3:::${SQLLayerManifestBucket}/${key}`;
 
   const resourceName = resourceNames.sqlLayerVersionResolverCustomResource;
+
+  // If deploying in a sandbox, use the same physical ID to speed up deployments
+  // Otherwise, make the physical ID change each time we do a deployment, so we always check for the latest version. This means we will
+  // never have a strictly no-op deployment, but the SQL Lambda configuration won't change unless the actual layer value changes
+  let physicalResourceId;
+  if (provideHotswapFriendlyResources(context)) {
```

Comment:
nit: rename this utility to match the flag behavior
```suggestion
  if (shouldProvisionHotswapFriendlyResources(context)) {
```